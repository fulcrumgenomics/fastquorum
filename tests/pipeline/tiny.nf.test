nextflow_pipeline {

    name "Test workflow - tiny"
    script "main.nf"
    tag "tiny"
    tag "pipeline"

    test("tiny") {

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.tiny.csv'
                fasta  = 'https://github.com/nf-core/test-datasets/raw/fastquorum/references/chr17.fa'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("fastquorum_mqc_versions") },
                { assert workflow.trace.succeeded().size() == 12 },
                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_report.html"),
                    path("$outputDir/consensus_filtering/filtered/SRR6109255/SRR6109255.cons.filtered.bam"),
                ).match("tiny") }
            )
        }

    }

}
