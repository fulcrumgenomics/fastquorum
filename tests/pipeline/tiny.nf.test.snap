{
    "tiny_rd_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLDDUPLEXCONSENSUSREADS={fgbio=2.0.2}, COLLECTDUPLEXSEQMETRICS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, FILTERCONSENSUSREADS={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0.0}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T15:35:22.555322"
    },
    "tiny_ht_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLANDFILTERDUPLEXCONSENSUSREADS={fgbio=2.0.2}, COLLECTDUPLEXSEQMETRICS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0.0}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T15:42:18.622029"
    },
    "tiny_rd": {
        "content": [
            "SRR6109255.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-19T23:42:19.171887"
    },
    "tiny_ht": {
        "content": [
            "SRR6109255.grouped-family-sizes.txt:md5,d2b8b8eb6c01a58f9aea6596b44e3735",
            "SRR6109255.duplex_seq_metrics.family_sizes.txt:md5,1439dc926491ef89f34dc5c56dc61b56"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-19T23:48:57.112733"
    }
}